clippy_lints/src/functions/mod.rs

```diff
@@ -354,6 +355,29 @@ declare_clippy_lint! {
     "`impl Trait` is used in the function's parameters"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Triggers when a function with a name ending with `_unchecked` is not marked as unsafe
+    /// ### Why is this bad?
+    /// `_unchecked` functions do not check some invariants which could result in business logic, memory safety or undefined behaviour if the user does not guarantee them.
```

Comment:
Many `_unchecked` functions do not have safety requirements, as logic errors are [not considered unsafe](https://doc.rust-lang.org/reference/behavior-not-considered-unsafe.html#logic-errors) I don't think we should be recommending using `unsafe` to denote that

Reply:
So maybe "`_unchecked` functions may result in memory safety or undefined behaviour, so marking them with unsafe is a way to alert of their uncertainty"?

Im not good at lint descriptions.